File: app/src/main/java/com/perflyst/twire/service/TempStorage.java
package com.perflyst.twire.service;

import com.perflyst.twire.model.ChannelInfo;

import java.util.Collection;
import java.util.concurrent.CopyOnWriteArrayList;

/**
 * Created by SebastianRask on 06-07-2015.
 */
public class TempStorage {
    private static CopyOnWriteArrayList<ChannelInfo> loaded_streamers;

    //
    // For StreamerInfo Objects
    //

    public static CopyOnWriteArrayList<ChannelInfo> getLoadedStreamers() {
        if (loaded_streamers == null)
            loaded_streamers = new CopyOnWriteArrayList<>();

        return loaded_streamers;
    }

    public static void addLoadedStreamer(ChannelInfo aStreamer) {
        getLoadedStreamers().addIfAbsent(aStreamer);
    }

    public static void addLoadedStreamer(Collection<ChannelInfo> aStreamerList) {
        getLoadedStreamers().addAllAbsent(aStreamerList);
    }

    public static void removeLoadedStreamer(ChannelInfo aStreamer) {
        getLoadedStreamers().remove(aStreamer);
    }

    public static boolean hasLoadedStreamers() {
        return getLoadedStreamers().size() > 0;
    }

    public static boolean containsLoadedStreamer(ChannelInfo aStreamer) {
        return getLoadedStreamers().contains(aStreamer);
    }

}

File: app/src/main/java/com/perflyst/twire/service/TempStorage.kt
package com.perflyst.twire.service

import com.perflyst.twire.model.ChannelInfo
import java.util.concurrent.CopyOnWriteArrayList

/**
 * Created by SebastianRask on 06-07-2015.
 */
object TempStorage {
    private var loaded_streamers: CopyOnWriteArrayList<ChannelInfo>? = null

    val loadedStreamers: CopyOnWriteArrayList<ChannelInfo>
        //
        get() {
            if (loaded_streamers == null) loaded_streamers =
                CopyOnWriteArrayList<ChannelInfo>()

            return loaded_streamers!!
        }

    fun addLoadedStreamer(aStreamer: ChannelInfo) {
        loadedStreamers.addIfAbsent(aStreamer)
    }

    fun addLoadedStreamer(aStreamerList: MutableCollection<ChannelInfo>) {
        loadedStreamers.addAllAbsent(aStreamerList)
    }

    fun removeLoadedStreamer(aStreamer: ChannelInfo) {
        loadedStreamers.remove(aStreamer)
    }

    fun hasLoadedStreamers(): Boolean {
        return !loadedStreamers.isEmpty()
    }

    fun containsLoadedStreamer(aStreamer: ChannelInfo): Boolean {
        return loadedStreamers.contains(aStreamer)
    }
}
